### ISSUE TITLE
Support zero-length `axis` in `.map_axis/_mut()`

### ISSUE THREAD

**Issue by jturner314** [rust-ndarray/ndarray#579]
The implementation of `.map_axis/_mut()` panics when the specified `axis` is zero-length. Ideally, we'd just apply `mapping` to empty views instead.

Example:

```rust
extern crate ndarray;

use ndarray::prelude::*;

fn main() {
    let a = Array2::<f32>::zeros((0, 2));
    a.map_axis(Axis(0), |_| {});
}
```

**Comment by andrei-papou** [comment]
Hi @jturner314 . I can work on this one if you don't mind.

**Comment by jturner314** [comment]
@andrei-papou Sure, please feel free.

**Comment by andrei-papou** [comment]
@jturner314 could you please point me to an example of creating an empty array? Thanks a lot.

**Comment by jturner314** [comment]
The simplest way to create a 1-dimensional, zero-length view in `map_axis` would be `unsafe { ArrayView::new_(std::ptr::NonNull::<A>::dangling().as_ptr(), Ix1(0), Ix1(0)) }`. See [this commentary](https://github.com/rust-ndarray/ndarray/blob/47b26916b242f1dd5ba92703c64e8c7e3d55caaa/src/lib.rs#L1130-L1136).

**Comment by andrei-papou** [comment]
@jturner314 thanks. please take a look at the PR https://github.com/rust-ndarray/ndarray/pull/612

**Comment by jturner314** [comment]
Fwiw, I realized that a cleaner (safe) way to create a zero-length one-dimensional view is `ArrayView1::from(&[])`.

**Comment by jblondin** [comment]
PR was merged. Is anything still remaining here, or can this issue be closed?

**Comment by jturner314** [comment]
Fixed by #612. Thanks for following up.
